### PR TITLE
Add an example of combining face detection and face alignment models

### PR DESCRIPTION
**Pull request by juliabeliaeva**
This is a demonstration of how to overcome the problem with the face alignment model described in #541 and is a part of fixing #544. Only works after merging #542 because of the problems with the dimensions order.

Result:
image

### CODE REVIEW

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    val faces = detectionModel.use { it.detectFaces(image) }
+
+    val alignmentModel = ONNXModels.FaceAlignment.Fan2d106.pretrainedModel(modelHub)
+    alignmentModel.printSummary()
+
+    val facesToLandmarks = alignmentModel.use {
+        faces.associateWith { face -> alignmentModel.predictOnCrop(image, face) }
+    }
+
+    val width = 600
```
Comment (zaleslaw) [inline examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:47]: Let's move to constants

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+private fun <T : FlatShape<T>> OnnxHighLevelModel<BufferedImage, List<T>>.predictOnCrop(image: BufferedImage,
+                                                                                        face: DetectedObject
+): List<T> {
+    val faceWidth = face.xMax - face.xMin
+    val faceHeight = face.yMax - face.yMin
+    val x1 = ((face.xMin - 0.1f * faceWidth) * image.width).toInt().coerceAtLeast(0)
+    val y1 = ((face.yMin - 0.1f * faceHeight) * image.height).toInt().coerceAtLeast(0)
+    val x2 = ((face.xMax + 0.1f * faceWidth) * image.width).toInt().coerceAtMost(image.width)
+    val y2 = ((face.yMax + 0.1f * faceHeight) * image.height).toInt().coerceAtMost(image.height)
+    val cropImage = pipeline<BufferedImage>().crop {
```
Comment (zaleslaw) [inline examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:67]: lets add a blank line

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        left = x1
+        top = y1
+        right = image.width - x2
+        bottom = image.height - y2
+
+    }.apply(image)
+    return predict(cropImage).map { shape ->
```
Comment (zaleslaw) [inline examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:74]: let's add a blank line again

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        outputWidth = width
+        outputHeight = width * image.height / image.width
+    }
+    showFrame(
+        "Detected Landmarks For ${facesToLandmarks.size} Face" + (if (facesToLandmarks.size == 1) "" else "s"),
+        createDetectedLandmarksPanel(resize.apply(image), facesToLandmarks.values.flatten())
+    )
+}
+
+private fun <T : FlatShape<T>> OnnxHighLevelModel<BufferedImage, List<T>>.predictOnCrop(image: BufferedImage,
```
Comment (zaleslaw) [inline examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:58]: cool function, let's add some notes about it and probably make public (give some hints to the user, if he want to copy our code)

How do you think, do we need this in our utils?

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    val image = ImageConverter.toBufferedImage(getFileFromResource("datasets/poses/multi/2.jpg"))
+
+    val modelHub = ONNXModelHub(cacheDirectory = File("cache/pretrainedModels"))
+
+    val detectionModel = ONNXModels.FaceDetection.UltraFace320.pretrainedModel(modelHub)
+    detectionModel.printSummary()
+
+    val faces = detectionModel.use { it.detectFaces(image) }
+
+    val alignmentModel = ONNXModels.FaceAlignment.Fan2d106.pretrainedModel(modelHub)
```
Comment (zaleslaw) [inline examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:40]: Sorry, I don't see where is closed this model?

**Review by zaleslaw — CHANGES_REQUESTED** [review]

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt (hunk 4 above)
Comment (juliabeliaeva) [reply examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:58]: Yes, it seems useful, I'll move it from the examples to a better location.

**Comment by juliabeliaeva** [comment]
> Also, the place is unclear—if it's not a runnable test, probably better to move into the visualization folder

It's not very different from other files in this package, like `Fan2D106LightAPIwithVisualization.kt`. Should all of the similar examples be moved?

File: examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt (hunk 5 above)
Comment (zaleslaw) [reply examples/src/main/kotlin/examples/onnx/faces/FaceDetectionAndAlignmentWithVisualization.kt:39]: haha, I missed the use usage, sorry

**Review by zaleslaw — APPROVED** [review]

**Pull request merged**